src/Generator/Passes/SymbolsCodeGenerator.cs

```diff
@@ -51,29 +51,91 @@ public override bool VisitMethodDecl(Method method)
 
             if (method.Namespace is ClassTemplateSpecialization &&
                 (method.TranslationUnit.IsSystemHeader ||
-                 ((method.IsConstructor || method.IsDestructor) &&
-                  !method.IsImplicit && !method.IsDefaulted && !method.IsPure &&
+                 (!method.IsImplicit && !method.IsDefaulted && !method.IsPure &&
                   string.IsNullOrEmpty(method.Body))))
             {
                 WriteLine($"template {GetExporting()}{method.Visit(cppTypePrinter)};");
                 return true;
             }
+
+            Class @class = (Class) method.Namespace;
+            bool needSubclass = method.Access == AccessSpecifier.Protected ||
+                @class.IsAbstract;
+            string wrapper = GetWrapper(method);
+            int i = 0;
+            foreach (var param in method.Parameters.Where(
+                p => string.IsNullOrEmpty(p.OriginalName)))
+                param.Name = "_" + i++;
+            var @params = string.Join(", ", method.Parameters.Select(CastIfRVReference));
+            if (needSubclass)
+            {
+                string className = @class.Visit(cppTypePrinter);
+                Write($"class {wrapper}{method.Namespace.Name} : public {className} {{ public: ");
+                Write(wrapper + method.Namespace.Name);
+                Write($@"({string.Join(", ", method.Parameters.Select(
+                    p => cppTypePrinter.VisitParameter(p)))})");
+                Write($": {className}({@params}) {{}}; ");
+                Method abstractDtor = null;
+                if (@class.IsAbstract)
+                {
+                    cppTypePrinter.MethodScopeKind = TypePrintScopeKind.Local;
+                    foreach (Method @abstract in @class.GetAbstractMethods())
+                    {
+                        if (@abstract.IsDestructor)
+                        {
+                            abstractDtor = @abstract;
+                            continue;
+                        }
+                        Write(@abstract.Visit(cppTypePrinter) + " {");
+                        Type returnType = @abstract.OriginalReturnType.Type.Desugar();
+                        if (!returnType.IsPrimitiveType(PrimitiveType.Void))
+                        {
+                            if (returnType.IsReference())
+                            {
+                                var pointee = returnType.GetPointee().Desugar().Visit(cppTypePrinter);
+                                Write($" static const auto __value = {pointee}{{}}; return __value; ");
```

Comment:
Not sure about the reason but it doesn't so let it stay, might generate less ASM or something.

Reply:
The new code no longer has the issue because you are using a C cast that basically says "I know what I'm doing" but it may be more prone to give warnings. What I meant before was this:

```.cpp
int& method() {
  static const int value; 
  return value; // error- cant remove const
}

const int& method() {
  static int value;
  return value; // ok - adding const is safe
}
```